### PR TITLE
Deprecate footnotes and removing wrap_text

### LINKED ISSUE #736: Should we deprecate `footnotes()` function in tern?
I am wondering if we need this functionality, considering that it can be added as an attribute directly if needed internally and the fact that largely overlaps with `rtables`' footnotes

_Originally posted by @Melkiades in https://github.com/insightsengineering/tern/pull/734#discussion_r1012630118_

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #736

### CODE REVIEW

File: R/g_forest.R [hunk 1]
```diff
@@ -246,12 +246,6 @@ g_forest <- function(tbl, # nolint
     vp = grid::plotViewport(margins = rep(1, 4))
   )
 
-  fn <- footnotes(tbl)
-  if (!is.null(fn)) {
-    footnotes(grob_forest) <- fn
-    message("grob footnote is not added to plot; suggest to use decorate_grob() to further decorate the grob")
-  }
-
```
Comment (Melkiades) [inline R/g_forest.R:254]: @shajoezhu this is never used imo and it is bringing forwards little information

File: R/footnotes.R [hunk 2]
```diff
@@ -12,14 +12,16 @@
 #' footnotes(x) <- "Species are equally distributed"
 #' attributes(x)
 `footnotes<-` <- function(x, value = NULL) { # nolint
+  dep_msg <- 'Please use current implementation in rtables or directly attr(x, "footnote")'
+  lifecycle::deprecate_warn("0.7.10", "footnotes()", details = dep_msg)
```
Comment (shajoezhu) [inline R/footnotes.R:16]: @pawelru , I was wondering should this be 0.7.11?

**Review by shajoezhu — COMMENTED** [review]
lgtm! Thanks @Melkiades. One minor question that I would like to check with Pawel with the version number, but otherwise everything is good. Thanks!

File: R/footnotes.R (hunk 2 above)
Comment (Melkiades) [reply R/footnotes.R:16]: I was wandering the same ahah

Comment (pawelru) [reply R/footnotes.R:16]: Yes you are right - this should point to the future version - 0.7.10 is the current one

**Comment by Melkiades** [comment]
@shajoezhu this is good to go ;)

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @Melkiades

**Pull request merged**